crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -682,7 +682,7 @@ declare_lint! {
 
 declare_lint! {
     /// ## What it does
-    /// Checks for `assert_type()` calls where the actual type
+    /// Checks for `assert_type()` and `assert_never()` calls where the actual type
```

Comment:
I did consider making more changes in the lint description here. I also considered a dedicated new diagnostic. But eventually thought that this might be enough? Let me know if you have a different opinion.